packages/gatsby/cache-dir/static-entry.js

```diff
@@ -168,8 +174,13 @@ export default (pagePath, callback) => {
   }
 
   const pageDataRaw = fs.readFileSync(getPageDataFile(pagePath))
-  const pageData = JSON.parse(pageDataRaw)
+  const pageData = JSON.parse(pageDataRaw.toString())
   const pageDataUrl = getPageDataUrl(pagePath)
+
+  const appDataRaw = fs.readFileSync(getAppDataFile())
+  const appData = JSON.parse(appDataRaw.toString())
+  const appDataUrl = getAppDataUrl()
```

Comment:
The app data stuff should is global, so we shouldn't have to re-read same file seperetaly for each html page - so I think we could move this outside of this function and have in module scope? Or alternatively have some memoization for it?

Reply:
That's a good idea